osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/JreEncryptionProvider.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.security.providers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.spec.AlgorithmParameterSpec;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import org.opensmartgridplatform.shared.security.EncryptionProviderType;
+
+public class JreEncryptionProvider extends AbstractEncryptionProvider implements EncryptionProvider {
+
+    public static final String DEFAULT_SINGLE_KEY_REFERENCE = "1";
+    public static final String ALG = "AES";
+    public static final String ALGORITHM = "AES/CBC/PKCS5PADDING";
+    public static final String PROVIDER = "SunJCE";
+    public static final String FORMAT = "RAW";
+    private static final byte[] IV = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+
+    private byte[] key;
+
+    protected Cipher getCipher() throws NoSuchPaddingException, NoSuchAlgorithmException, NoSuchProviderException {
+        return Cipher.getInstance(ALGORITHM, PROVIDER);
+    }
+
+    public JreEncryptionProvider(File keyStoreFile) {
+        try {
+            super.setKeyFile(keyStoreFile);
+            this.key = Files.readAllBytes(Paths.get(keyStoreFile.getAbsolutePath()));
+        } catch (IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)